Fix for using system hostname as default - ID: 1145855
James P. Dugal ( jpd2 ) - 2005-02-21 23:42
http://sourceforge.net/tracker/?func=detail&aid=1145855&group_id=63352&atid=503661

diff -BurpN gateway/rp-l2tp-0.4/tunnel.c gateway/rp-l2tp/tunnel.c
--- gateway/rp-l2tp-0.4/tunnel.c	2009-09-02 16:56:19.000000000 +0000
+++ gateway/rp-l2tp/tunnel.c	2009-09-02 17:00:06.000000000 +0000
@@ -586,7 +586,7 @@ tunnel_send_SCCRQ(l2tp_tunnel *tunnel)
     l2tp_dgram_add_avp(dgram, tunnel, MANDATORY,
 		  sizeof(u32), VENDOR_IETF, AVP_FRAMING_CAPABILITIES, &u32);
 
-    hostname = tunnel->peer->hostname ? tunnel->peer->hostname : Hostname;
+    hostname = tunnel->peer->hostname[0] ? tunnel->peer->hostname : Hostname;
 
     /* Host name */
     l2tp_dgram_add_avp(dgram, tunnel, MANDATORY,
@@ -808,7 +808,7 @@ tunnel_handle_SCCRQ(l2tp_dgram *dgram,
     l2tp_dgram_add_avp(dgram, tunnel, MANDATORY,
 		  sizeof(u32), VENDOR_IETF, AVP_FRAMING_CAPABILITIES, &u32);
 
-    hostname = tunnel->peer->hostname ? tunnel->peer->hostname : Hostname;
+    hostname = tunnel->peer->hostname[0] ? tunnel->peer->hostname : Hostname;
 
     /* Host name */
     l2tp_dgram_add_avp(dgram, tunnel, MANDATORY,
